js/lykke2.js

```diff
@@ -0,0 +1,1174 @@
+    }
+
+    async fetchDepositAddress (code, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const request = {
+            'AssetId': this.safeString (currency, 'id'),
+        };
+        const response = await this.privateGetOperationsDepositsAddressesAssetId (this.extend (request, params));
+        //
+        //     {
+        //         "assetId":"2a34d6a6-5839-40e5-836f-c1178fa09b89",
+        //         "symbol":"BCH",
+        //         "address":null,
+        //         "baseAddress":null,
+        //         "addressExtension":null,
+        //         "state":"Active"
+        //     }
+        //
+        const address = this.safeString (response, 'baseAddress');
+        const tag = this.safeString (response, 'addressExtension');
+        this.checkAddress (address);
+        return {
+            'currency': code,
+            'address': address,
+            'tag': tag,
+            'network': undefined,
+            'info': response,
+        };
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        //
+        // withdraw
+        //     "3035b1ad-2005-4587-a986-1f7966be78e0"
+        //
+        // fetchTransactions
+        //     {
+        //         "operationId":"787201c8-f1cc-45c0-aec1-fa06eeea426b",
+        //         "assetId":"2a34d6a6-5839-40e5-836f-c1178fa09b89",
+        //         "totalVolume":0.1,
+        //         "fee":0.0,
+        //         "type":"deposit",
+        //         "timestamp":1644146723620
+        //     }
+        //
+        let id = undefined;
+        let assetId = undefined;
+        let code = undefined;
+        let amount = undefined;
+        let fee = undefined;
+        let type = undefined;
+        let timestamp = undefined;
+        if (typeof transaction === 'string') {
+            id = transaction;
+        } else {
+            id = this.safeString (transaction, 'operationId');
+            assetId = this.safeString (transaction, 'assetId');
+            code = this.safeCurrencyCode (assetId, currency);
+            amount = this.safeNumber (transaction, 'totalVolume');
+            fee = this.safeNumber (transaction, 'fee');
+            type = this.safeString (transaction, 'type');
+            timestamp = this.safeInteger (transaction, 'timestamp');
+        }
+        return {
+            'info': transaction,
+            'id': id,
+            'txid': undefined,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'network': undefined,
+            'addressFrom': undefined,
+            'address': undefined,
+            'addressTo': undefined,
+            'tagFrom': undefined,
+            'tag': undefined,
+            'tagTo': undefined,
+            'type': type,
+            'amount': amount,
+            'currency': code,
+            'status': undefined,
+            'updated': undefined,
+            'fee': {
```

Comment:
We should not fill this structure, if `feeCost` is undefined, see how that is done in the other implementations of parseTransaction:

- https://github.com/ccxt/ccxt/blob/master/js/huobi.js#L4093
- https://github.com/ccxt/ccxt/blob/master/js/mexc.js#L1501
- https://github.com/ccxt/ccxt/blob/master/js/hitbtc3.js#L949